src/System.Management.Automation/engine/hostifaces/LocalConnection.cs

```diff
@@ -47,6 +47,10 @@ internal sealed partial class LocalRunspace : RunspaceBase
         internal LocalRunspace(PSHost host, InitialSessionState initialSessionState, bool suppressClone)
             : base(host, initialSessionState, suppressClone)
         {
+            if (RemoteSessionNamedPipeServer.IPCNamedPipeServer == null)
```

Comment:
I am worried about this because `CreateIPCNamedPipeServerSingleton` is called in a static constructor of RemoteSessionNamedPipeServer, so this creates a potential race condition.  I think `CreateIPCNamedPipeServerSingleton` will need to **force** the creation of a server object with the provided pipe name, in case the static constructor is run first.

Also, we don't need the null check.

Reply:
I removed it from the static constructor and moved it here so that I could access the `InitialSessionState` of `LocalRunspace`.

Reply:
I feel this is the wrong approach.  This default IPC listener should not be tied to the creation of a local runspace, and should remain as it is now.  I think it is fine to start the powershell.exe host with a switch specifying a custom pipe name for special applications like PowerShell VSCode extension, but it should either replace the default IPC or run as an optional second IPC channel.

Reply:
I see what you're saying. 

The problem I faced was that I didn't see the clearest route for grabbing the pipe name from the pwsh.exe parameters to the static constructor.

Going through the `initialsessionstate`, to the runspace seemed like the best option because it would be easily exposed in the PowerShell SDK. You could set the `PipeName` in the `InitialSessionState` and get this feature.

Whats your recommendation for implementing this such that it could also be easily exposed in the SDK as well as via `pwsh.exe`?